R/dataset.R

```diff
@@ -88,6 +88,80 @@ setMethod("notes<-", "CrunchDataset", function (x, value) {
     invisible(setEntitySlot(x, "notes", value))
 })
 
+#' Get and set the market size for Crunch datasets
+#'
+#' Crunch Datasets allow you to set a target population size in order to extrapolate
+#' population estimates from survey percentages. This function lets you work with
+#' the population size and magnitude.
+#'
+#' @param x a Crunch Dataset
+#' @param value For the setters, the `size` or `magnitude` to be set
+#' @param size the target population size, to remove a population set to `NULL`
+#' @param magnitude the order of magnitude with which to display the population
+#' size. Must be either `3`, `6`, or `9` for thousands, millions, and billions respectively.
+#' @return `popSize` and `popMagnitude` return the population size or
+#' magnitude. `setPopulation` returns the modified dataset.
+#' @name population
+#' @aliases popSize popMagnitude setPopulation popSize<- popMagnitude<-
+NULL
+
+#' @rdname population
+#' @export
+setMethod("popSize", "CrunchDataset", function (x) {
+    size <- settings(x)$population$size
+    if (size == 0) {
+        return(NULL)
+    }
+    return(size) #the API value 0 stands for "No Population set"
+})
+
+#' @rdname population
+#' @export
+
+setMethod("popSize<-", "CrunchDataset", function (x, value) {
+    setPopulation(x, size = value)
+})
+
+#' @rdname population
+#' @export
+setMethod("popMagnitude", "CrunchDataset", function (x) {
+    return(settings(x)$population$magnitude)
+})
+
+#' @rdname population
+#' @export
+setMethod("popMagnitude<-", "CrunchDataset", function (x, value) {
+   setPopulation(x, magnitude = value)
+})
+
+#' @rdname population
+#' @export
+setMethod("setPopulation", "CrunchDataset", function (x, size, magnitude) {
+    pop <- settings(x)$population
+    if (missing(magnitude)) {
+        if (is.null(pop$magnitude)) {
+            warning("Dataset magnitude not set, defaulting to thousands")
+            magnitude <- 3
+        } else {
+            magnitude <- pop$magnitude
+        }
+    }
+    if (missing(size)) {
+        if (is.null(pop$size)) {
+            halt("Dataset does not have a population, please set one before attempting to change magnitude")
+        }
+        size <- pop$size
+    }
+    if (is.null(size)) {
+        size <- 0
+    }
```

Comment:
If the size is being set to `NULL` there should be no `size` and no `magnitude` attributes but rather `population` should be `NULL`. That's what whaam does when population is removed and is more in line with other settings. This might be more easily accomplished by making a method with `setMethod("setPopulation<-", c("CrunchDataset", "NULL"), function ...`

An example of something that is similar is geography metadata: a variable that has a attached geography information will have a `view.geodata` attribute that includes things like `geodatum` `feature_key` etc. but when you set that to `NULL` in R, it empties out `view.geodata` to be empty: https://github.com/Crunch-io/rcrunch/blob/master/R/geo.R#L63-L70 rather than just deleting the individual `geodatum` and `feature_key` attributes.